docs/porting/custom-target-porting.md

```diff
@@ -0,0 +1,259 @@
+
+   If you don't already have an Mbed program on your computer, at a command terminal, run this Mbed CLI command:
+
+   ```
+   mbed new --program mbed-os-imaginary-port
+   ```
+  
+   This command imports `mbed-os` from the [official Mbed OS source repository](https://github.com/armmbed/mbed-os) into a new directory called `mbed-os-imaginary-port`.
+
+   Now, change directories into your new project:  
+
+  ```
+  cd mbed-os-imaginary-port
+  ```
+
+1. Create a new file named `custom_targets.json` at the same level as the `mbed-os` directory:
+
+   Inspect the contents of `mbed-os/targets/targets.json`. For this example, search for `DISCO_L475VG_IOT01A`.
+
+   Copy the contents from the `DISCO_L475VG_IOT01A` section into your `custom_targets.json` file. Be sure to include brackets `{ }` surrounding the content.
+
+   Then make changes for your board. For example, after making changes, the full contents look like this:
+   
+   ```
+   {
+     "IMAGINARYBOARD": {
+         "components_add": ["QSPIF", "FLASHIAP"],
+         "inherits": ["FAMILY_STM32"],
+         "core": "Cortex-M4F",
+         "extra_labels_add": ["STM32L4", "STM32L475xG", "STM32L475VG"],
+         "config": {
+             "clock_source": {
+                 "help": "Mask value : USE_PLL_HSE_EXTC (need HW patch) | USE_PLL_HSE_XTAL (need HW patch) | USE_PLL_HSI | USE_PLL_MSI",
+                 "value": "USE_PLL_MSI",
+                 "macro_name": "CLOCK_SOURCE"
+             },
+             "lpticker_lptim": {
+                 "help": "This target supports LPTIM. Set value 1 to use LPTIM for LPTICKER, or 0 to use RTC wakeup timer",
+                 "value": 1
+             }
+         },
+         "overrides": { "lpticker_delay_ticks": 4 },
+         "detect_code": ["1234"],
+         "macros_add": [
+             "MBED_TICKLESS",
+             "MBED_SPLIT_HEAP"
+         ],
+         "device_has_add": [
+             "CRC",
+             "TRNG",
+             "FLASH",
+             "QSPI",
+             "MPU"
+         ],
+         "device_has_remove": [         
+             "ANALOGIN",
+             "I2CSLAVE",
+             "I2C_ASYNCH"
+         ],                     
+         "release_versions": ["2", "5"],
+         "device_name": "STM32L475VG",
+         "bootloader_supported": true
+     }
+   }
+   ```
+
+   Let's review the changes one by one.
+
+   - The board name was changed from `DISCO_L475VG_IOT01A` to `IMAGINARYBOARD` so the board can be uniquely identified.
+
+   - The `detect_code` was changed from `0764` to `1234`. The `detect_code` is a unique number that identifies the board to the Mbed OS test tools. This number is typically built into the debug interface.
+
+   - The `macros_add` section was changed to remove `USBHOST_OTHER` because USB is not used on the new board.
+
+   - The `device_has_add` section was changed to not add the `ANALOGOUT`, `CAN`, `USBDEVICE` drivers because those features are not used on the new board.
```

Comment:
```suggestion
   - The `device_has_add` section was changed to not add the `ANALOGOUT`, `CAN`, and `USBDEVICE` drivers because those features are not used on the new board.
```